workflow_carbon_tracker.ipynb

```diff
@@ -13,2717 +13,117 @@
     "- ERA5-land (monthly)\n",
```

Comment:
<p>I got an error when running this cell:</p><p> </p><p><code></code><code>py</p><p>---------------------------------------------------------------------------</p><p>TypeError                                 Traceback (most recent call last)</p><p>File <a href="https://vscode-remote+wsl-002bubuntu-002d20-002e04.vscode-resource.vscode-cdn.net/home/yangliu/Excited/excited-workflow/~/venv/excited/lib/python3.10/site-packages/pandas/core/indexes/base.py:3391" target="_blank">~/venv/excited/lib/python3.10/site-packages/pandas/core/indexes/base.py:3391</a>, in Index._union(self, other, sort)</p><p><strong>   </strong><a href="https://vscode-remote+wsl-002bubuntu-002d20-002e04.vscode-resource.vscode-cdn.net/home/yangliu/Excited/excited-workflow/~/venv/excited/lib/python3.10/site-packages/pandas/core/indexes/base.py:3390" target="_blank"><strong>3390</strong></a><strong> try</strong>:</p><p>-&gt; <a href="https://vscode-remote+wsl-002bubuntu-002d20-002e04.vscode-resource.vscode-cdn.net/home/yangliu/Excited/excited-workflow/~/venv/excited/lib/python3.10/site-packages/pandas/core/indexes/base.py:3391" target="_blank">3391</a>     return self._outer_indexer(other)[0]</p><p><strong>   </strong><a href="https://vscode-remote+wsl-002bubuntu-002d20-002e04.vscode-resource.vscode-cdn.net/home/yangliu/Excited/excited-workflow/~/venv/excited/lib/python3.10/site-packages/pandas/core/indexes/base.py:3392" target="_blank"><strong>3392</strong></a><strong> except</strong> (TypeError, IncompatibleFrequency):</p><p><strong>   </strong><a href="https://vscode-remote+wsl-002bubuntu-002d20-002e04.vscode-resource.vscode-cdn.net/home/yangliu/Excited/excited-workflow/~/venv/excited/lib/python3.10/site-packages/pandas/core/indexes/base.py:3393" target="_blank"><strong>3393</strong></a><strong>     # incomparable objects; should only be for object dtype</strong></p><p> </p><p>File <a href="https://vscode-remote+wsl-002bubuntu-002d20-002e04.vscode-resource.vscode-cdn.net/home/yangliu/Excited/excited-workflow/~/venv/excited/lib/python3.10/site-packages/pandas/core/indexes/base.py:387" target="_blank">~/venv/excited/lib/python3.10/site-packages/pandas/core/indexes/base.py:387</a>, in Index._outer_indexer(self, other)</p><p><strong>    </strong><a href="https://vscode-remote+wsl-002bubuntu-002d20-002e04.vscode-resource.vscode-cdn.net/home/yangliu/Excited/excited-workflow/~/venv/excited/lib/python3.10/site-packages/pandas/core/indexes/base.py:386" target="_blank"><strong>386</strong></a><strong> ov = cast(np.ndarray, ov)</strong></p><p>--&gt; <a href="https://vscode-remote+wsl-002bubuntu-002d20-002e04.vscode-resource.vscode-cdn.net/home/yangliu/Excited/excited-workflow/~/venv/excited/lib/python3.10/site-packages/pandas/core/indexes/base.py:387" target="_blank">387</a> joined_ndarray, lidx, ridx = libjoin.outer_join_indexer(sv, ov)</p><p><strong>    </strong><a href="https://vscode-remote+wsl-002bubuntu-002d20-002e04.vscode-resource.vscode-cdn.net/home/yangliu/Excited/excited-workflow/~/venv/excited/lib/python3.10/site-packages/pandas/core/indexes/base.py:388" target="_blank"><strong>388</strong></a><strong> joined = self._from_join_target(joined_ndarray)</strong></p><p> </p><p>File <a href="https://vscode-remote+wsl-002bubuntu-002d20-002e04.vscode-resource.vscode-cdn.net/home/yangliu/Excited/excited-workflow/~/venv/excited/lib/python3.10/site-packages/pandas/_libs/join.pyx:564" target="_blank">~/venv/excited/lib/python3.10/site-packages/pandas/_libs/join.pyx:564</a>, in pandas._libs.join.outer_join_indexer()</p><p> </p><p>File src/cftime/_cftime.pyx:1432, in cftime._cftime.datetime.__richcmp__()</p><p> </p><p>TypeError: cannot compare cftime.DatetimeNoLeap(2000, 7, 2, 12, 0, 0, 0, has_year_zero=True) and Timestamp('1991-01-01 06:00:00', freq='H') (different calendars)</p><p> </p><p>During handling of the above exception, another exception occurred:</p><p> </p><p>TypeError                                 Traceback (most recent call last)</p><p><strong>/home/yangliu/Excited/excited-workflow/workflow_carbon_tracker.ipynb Cell 10</strong> line 9</p><p><strong>      </strong><a href="about:blank" target="_blank"><strong>1</strong></a> desired_data = [</p><p><strong>      </strong><a href="about:blank" target="_blank"><strong>2</strong></a>     "biomass",</p><p><strong>      </strong><a href="about:blank" target="_blank"><strong>3</strong></a>     "spei",</p><p>   (...)</p><p><strong>      </strong><a href="about:blank" target="_blank"><strong>6</strong></a>     "era5_land_monthly"</p><p><strong>      </strong><a href="about:blank" target="_blank"><strong>7</strong></a> ]</p><p><strong>      </strong><a href="about:blank" target="_blank"><strong>8</strong></a> ds_input = xr.merge(</p><p>----&gt; <a href="about:blank" target="_blank">9</a>     [datasets[name].load(freq="monthly", target_grid=ds_cb) for name in desired_data]</p><p><strong>     </strong><a href="about:blank" target="_blank"><strong>10</strong></a> )</p><p> </p><p><strong>/home/yangliu/Excited/excited-workflow/workflow_carbon_tracker.ipynb Cell 10</strong> line 9</p><p><strong>      </strong><a href="about:blank" target="_blank"><strong>1</strong></a> desired_data = [</p><p><strong>      </strong><a href="about:blank" target="_blank"><strong>2</strong></a>     "biomass",</p><p><strong>      </strong><a href="about:blank" target="_blank"><strong>3</strong></a>     "spei",</p><p>   (...)</p><p><strong>      </strong><a href="about:blank" target="_blank"><strong>6</strong></a>     "era5_land_monthly"</p><p><strong>      </strong><a href="about:blank" target="_blank"><strong>7</strong></a> ]</p><p><strong>      </strong><a href="about:blank" target="_blank"><strong>8</strong></a> ds_input = xr.merge(</p><p>----&gt; <a href="about:blank" target="_blank">9</a>     [datasets[name].load(freq="monthly", target_grid=ds_cb) for name in desired_data]</p><p><strong>     </strong><a href="about:blank" target="_blank"><strong>10</strong></a> )</p><p> </p><p>File <a href="https://vscode-remote+wsl-002bubuntu-002d20-002e04.vscode-resource.vscode-cdn.net/home/yangliu/Excited/excited-workflow/~/Excited/excited-workflow/src/excited_workflow/source_datasets/biomass.py:55" target="_blank">~/Excited/excited-workflow/src/excited_workflow/source_datasets/biomass.py:55</a>, in Biomass.load(self, freq, variables, target_grid)</p><p>...</p><p><strong>   </strong><a href="https://vscode-remote+wsl-002bubuntu-002d20-002e04.vscode-resource.vscode-cdn.net/home/yangliu/Excited/excited-workflow/~/venv/excited/lib/python3.10/site-packages/pandas/core/indexes/base.py:3400" target="_blank"><strong>3400</strong></a><strong> return</strong> np.array(value_list, dtype=object)</p><p> </p><p>File src/cftime/_cftime.pyx:1432, in cftime._cftime.datetime.__richcmp__()</p><p> </p><p>TypeError: cannot compare cftime.DatetimeNoLeap(2000, 7, 2, 12, 0, 0, 0, has_year_zero=True) and Timestamp('2000-07-02 12:00:00', freq='H') (different calendars)</p><p></code><code></code></p><p> </p><p>It has something to do with biomass dataset. It could be the case that I'm not using the correct data. I didn't have time today and will look into it tomorrow.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/EXCITED-CO2/excited-workflow/pull/30/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Reply:
😓 Just ignore the formatting...stupid reviewnb 😞 .

Reply:
Odd error...

> It could be the case that I'm not using the correct data. 

No this should go fine if you tested it on SRC. You'd use the config file I put in /etc/excited.

Perhaps it's an environment issue. Let's look at it together today.

Reply:
@ClaireDons was able to run the notebook fine yesterday.

Reply:
Yes, it works on SRC for me as well.